### PR TITLE
Add org.duckstation.DuckStation

### PR DESCRIPTION
**Pull request by j8r**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** @stenzek's approval: https://github.com/stenzek/duckstation/issues/1987
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

---

## Notes

~I didn't use `cmake-ninja` directly, because an `install` target is required but missing in the project (I'm no familiar with CMake to fix that (if in case if wanted by the upstream project)~ (thanks @hfiguiere https://github.com/flathub/flathub/pull/2306#discussion_r624793955)

## Permissions

Usual permissions for an emulator: Wayland/X11 + audio access for the UI, filesystem to load roms and BIOS.
Not sure about `network`: can DuckStation use it @stenzek?

### CODE REVIEW

**Comment by hfiguiere** [comment]
bot, build org.duckstation.DuckStation

File: org.duckstation.DuckStation.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      - /include
+      - /lib/pkgconfig
+      - /share
+    sources:
+      - type: archive
+        url: https://www.freedesktop.org/software/libevdev/libevdev-1.11.0.tar.xz
+        sha256: 63f4ea1489858a109080e0b40bd43e4e0903a1e12ea888d581db8c495747c2d0
+
+  - name: duckstation
+    no-make-install: true
```
Comment (hfiguiere) [inline org.duckstation.DuckStation.yml:32]: you forgot `buildsystem: cmake-ninja` here. This is what cause the build failure.

**Comment by j8r** [comment]
bot, build org.duckstation.DuckStation

**Comment by hfiguiere** [comment]
it's lacking a 128x128 icon

**Comment by j8r** [comment]
Yeah, I see the build log. @stenzek do you have such icon around? If you have the original "bigduck" image, it can be resized to 128, 256 and 512.

**Comment by hfiguiere** [comment]
if you have the SVG, you can use that.

File: flathub.json [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+  "publish-delay-hours": 0
```
Comment (barthalion) [inline flathub.json:2]: It doesn't do what you think it does, I'd just drop it. It's not critical to have it up to date immediately.

File: org.duckstation.DuckStation.yml [hunk 3]
```diff
@@ -0,0 +1,55 @@
+app-id: org.duckstation.DuckStation
+runtime: org.kde.Platform
+runtime-version: '5.15'
+sdk: org.kde.Sdk
+command: duckstation-nogui
+finish-args:
+  - --device=all
+  - --filesystem=host:ro
```
Comment (barthalion) [inline org.duckstation.DuckStation.yml:8]: Does it respect the XDG spec to save settings and save files in a writable directory?

Comment (j8r) [reply org.duckstation.DuckStation.yml:8]: Seems so: https://github.com/stenzek/duckstation/blob/56243a26ec7ceffa3a22cd9a4602d62e7330121a/src/frontend-common/common_host_interface.cpp#L935, `XDG_DATA_HOME` is used.

Comment (barthalion) [reply org.duckstation.DuckStation.yml:8]: Neat, thanks.

File: .github/workflows/nightly.yml [hunk 4]
```diff
@@ -0,0 +1,22 @@
+name: Update commit hash
```
Comment (bilelmoussaoui) [inline .github/workflows/nightly.yml:1]: does this work? can't https://github.com/flathub/flatpak-external-data-checker be used here?

File: org.duckstation.DuckStation.yml [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+      - -DUSE_DRMKMS=1
+    post-install:
+      - install -m 755 -t /app/bin/ bin/duckstation-nogui bin/duckstation-qt
+      - |-
+        for px in 16 32 48 64; do
+          install -Dm644 ../dist/icon-${px}px.png /app/share/icons/hicolor/${px}x${px}/apps/org.duckstation.DuckStation.png
+        done
+      - install -Dm644 ../dist/duckstation-nogui.desktop /app/share/applications/org.duckstation.DuckStation.desktop
+      - install -Dm644 ../dist/duckstation-qt.desktop /app/share/applications/org.duckstation.DuckStation.Qt.desktop
+      - sed -i 's/^Icon=.*/Icon=org.duckstation.DuckStation/' /app/share/applications/*.desktop
```
Comment (bilelmoussaoui) [inline org.duckstation.DuckStation.yml:48]: I would use desktop-file-edit here

File: .github/workflows/nightly.yml (hunk 4 above)
Comment (j8r) [reply .github/workflows/nightly.yml:1]: It works on Flycast, to be confirmed with this project.
As @barthalion [said](https://github.com/flathub/flathub/issues/1582#issuecomment-826522418), it likely better to have a simple, independent CRON for projects. The commit hash is supposed to change frequently.

Comment (j8r) [reply .github/workflows/nightly.yml:1]: It is also not well documented, for instance a `x-checker-data` comment is needed in the file to make it work. I prefer to have something simple, in control and independent.

File: org.duckstation.DuckStation.yml (hunk 5 above)
Comment (j8r) [reply org.duckstation.DuckStation.yml:48]: `Only one desktop file can be processed at once.`, and the result would be the same.

**Comment by j8r** [comment]
bot, build org.duckstation.DuckStation

**Pull request closed without merging**